java-checks/src/main/java/org/sonar/java/checks/ThreadOverridesRunCheck.java

```diff
@@ -72,4 +87,45 @@ private static boolean overridesRunMethod(Symbol.TypeSymbol classSymbol) {
     }
     return overridesRunMethod;
   }
+
+  private static boolean hasConstructorCallingSuperWithRunnable(ClassTree classTree) {
+    return classTree.members().stream()
+      .filter(member -> member.is(Tree.Kind.CONSTRUCTOR))
+      .map(MethodTree.class::cast)
+      .anyMatch(ThreadOverridesRunCheck::hasCallToSuperWithRunnable);
+  }
+
+  private static boolean hasRunnableArgument(Arguments args) {
+    return args.stream().anyMatch(ThreadOverridesRunCheck::isRunnable);
+  }
+
+  private static boolean isRunnable(ExpressionTree arg) {
+    return arg.symbolType().isSubtypeOf("java.lang.Runnable");
+  }
+
+  private static boolean hasCallToSuperWithRunnable(MethodTree constructor) {
+    SuperRunnableVisitor visitor = new SuperRunnableVisitor();
+    constructor.accept(visitor);
+    return visitor.callSuperWithRunnable;
+  }
+
+  private static class SuperRunnableVisitor extends BaseTreeVisitor {
+
+    private boolean callSuperWithRunnable = false;
+
+    private static final MethodMatcher SUPER_THREAD = MethodMatcher.create().typeDefinition("java.lang.Thread").name("<init>").withAnyParameters();
+
+    @Override
+    public void visitMethodInvocation(MethodInvocationTree tree) {
+      if (SUPER_THREAD.matches(tree) && ThreadOverridesRunCheck.hasRunnableArgument(tree.arguments())) {
+        callSuperWithRunnable = true;
+      }
+      super.visitMethodInvocation(tree);
+    }
+
+    @Override
+    public void visitClass(ClassTree tree) {
+      // skip inner classes
```

Comment:
There is one. [Line#47](https://github.com/SonarSource/sonar-java/pull/1589/files#diff-d93a2ec6a1ed7e239cc98988f8d389b5R47) would raise two issue if classes were not skipped.